packages/ra-core/src/store/localStorageStore.ts

```diff
@@ -116,15 +116,11 @@ export const localStorageStore = (
         },
         reset(): void {
             const storage = getStorage();
-            for (let i = 0; i < storage.length; i++) {
-                if (storage.key(i)?.substring(0, prefixLength) === prefix) {
-                    const key = storage.key(i)?.substring(prefixLength + 1);
-                    if (!key || !storage.key(i)) return;
-                    // @ts-ignore
-                    storage.removeItem(storage.key(i));
-                    publish(key, undefined);
-                }
-            }
+            Object.keys(storage).forEach(key => {
+                storage.removeItem(key);
+                const publishKey = key.substring(prefixLength + 1);
+                publish(publishKey, undefined);
+            });
```

Comment:
I agree with the analysis and the fix
But why did you also remove the `if (storage.key(i)?.substring(0, prefixLength) === prefix)` part? This should be kept IMO